2context.vim

```diff
@@ -8,6 +8,36 @@
 " output buffer. The script parses content line-by-line from the first buffer
 " and pastes the modified result on the second buffer.
 
+"" before starting open buffer to get syntaxid of vimTypingTeXTags syntax
+"" group. knowing the id simplifies and speedsup location of /BTEX /ETEX
+"" syntax group added by vimtypingfilter.vim syntax file.
+"" These are used to hide context commands from vim syntax highlighting and
+"" checking. For details see vim/syntax/vimtypingfilter.vim file
+
+"" open hidden [scratch] buffer
+new 
+setlocal buftype=nofile noswapfile bufhidden=hide
+
+"" write line containing begin and end tex tags only
+call setline(1,"/BTEX /ETEX")
+
+"" denominate vimtypingfilter as the syntax file to be used for this buffer
+set syntax=vimtypingfilter
+
+"" loop through characters of first line containing the string unti
+"" synid returns the id of the transparent 'vimTypingTeXTags' group
+"" stop as soon as found and remove buffer
+let s:vimtypingid = synID(1,0,0)
+let s:col = 1
+while synIDattr(s:vimtypingid,"name") != "vimTypingTeXTags"
+	let s:vimtypingid = synID(1,s:col,0)
+	let s:col = s:col + 1
+endwhile
+"" done discard the [scratch] buffer again
+q!
```

Comment:
Can't this be done simply using `let s:vimtypingid = hlID("vimTypingTeXTags")`?

Reply:
Oh, cool shure if that spares the use of an additional buffer than this is definitily the better solution. 
Seems as if i missed or didnt get something from the vim docs.